src/main/java/reposense/parser/AuthorConfigLocationParser.java

```diff
@@ -0,0 +1,130 @@
+     * into a 2-element list containing location and branch information.
+     * The first item in the list is the location, while the second item is the default branch
+     * from {@code author-config.csv} branch column.
+     */
+    private static List<String> parseDefault(String location, String defaultSpecifiedBranch) {
+        List<String> parsedLocationAndBranch = new ArrayList<>();
+        parsedLocationAndBranch.add(location);
+        parsedLocationAndBranch.add(defaultSpecifiedBranch);
+        return parsedLocationAndBranch;
+    }
+
+    /**
+     * Returns true if the given {@code locationAndBranch} is using the location and
+     * branch delimiter syntax. If the given {@code locationAndBranch} is an actual
+     * path, then it is considered to not be using the delimiter syntax.
+     */
+    private static boolean isUsingDelimiter(String locationAndBranch) {
+        int lastLocationBranchDelimiterIndex = locationAndBranch.lastIndexOf(LOCATION_BRANCHES_DELIMITER);
+        if (lastLocationBranchDelimiterIndex == -1) {
+            return false;
+        }
+
+        return !pathExists(locationAndBranch);
+    }
+
+    /**
+     * Parses a repository location into a list containing location and branch information.
+     * The first item in the list is the location, while subsequent items are the
+     * branches specified.
+     * <br><br>
+     * The repository location can contain several ~ or | characters as well. The last ~ character will
+     * be used as the separator between the actual path and the specified branches.
+     * <br><br>
+     * Example: home~Desktop~RepoSense~master|cypress is parsed to give [home~Desktop~RepoSense, master, cypress]
+     * <br><br>
+     * There exist an obscure case where the delimiter syntax will not work due to conflicts in directory names.
+     * See: <a href="https://github.com/reposense/RepoSense/pull/1961#discussion_r1155475445">PR #1961 Discussion</a>
+     */
+    private static List<String> parseDelimiter(String locationAndBranches) {
+        List<String> parsedLocationAndBranches = new ArrayList<>();
+        int lastLocationBranchesDelimiterIndex = locationAndBranches.lastIndexOf(
+                LOCATION_BRANCHES_DELIMITER);
+
+        String location = locationAndBranches.substring(0, lastLocationBranchesDelimiterIndex);
+        parsedLocationAndBranches.add(location);
+
+        String[] branches = locationAndBranches
+                .substring(lastLocationBranchesDelimiterIndex + 1).split(SPLIT_SAFE_BRANCHES_DELIMITER);
+        for (String branch : branches) {
+            parsedLocationAndBranches.add(branch);
+        }
+
+        if (branches.length == 0) {
+            // by default, if no branches are specified, an empty branch is returned since
+            // the only case where branches.length is 0 is when delimiter is used but no
+            // branch is specified.
+            parsedLocationAndBranches.add("");
```

Comment:
I was wondering why codecov didnt detect this so I went to check. As it turns out, even if no branches are specified, after the split, the last entry will be an empty string so the previous for loop adds an empty string. So I don't actually think this part is necessary

Reply:
Strangely enough, I recall I had a particular edge case that I was trying to resolve here, but I am unable to recall the exact case here. I will remove it for the time being, at least until if this edge case ever shows up (which is very unlikely).